ReactWindows/ReactNative.Shared/UIManager/ViewManagerBase.cs

```diff
@@ -247,14 +248,14 @@ void IViewManager.UpdateProps(object viewToUpdate, JObject props)
             OnAfterUpdateTransaction((TView)viewToUpdate);
         }
 
-        object IViewManager.CreateView(ThemedReactContext reactContext)
+        object IViewManager.CreateView(IReactContext reactContext)
         {
-            return CreateView(reactContext);
+            return CreateView((ThemedReactContext)reactContext);
         }
 
-        void IViewManager.OnDropViewInstance(ThemedReactContext reactContext, object view)
+        void IViewManager.OnDropViewInstance(IReactContext reactContext, object view)
         {
-            OnDropViewInstance(reactContext, (TView)view);
+            OnDropViewInstance((ThemedReactContext)reactContext, (TView)view);
```

Comment:
I think of `IViewManager` as an interface that you could implement if you wanted to, but it's not likely that you'll do it well. Honestly, once we move this to an interfaces project, we can probably put the [EditorBrowsable(Never)] attribute so folks can't really use it. The expectation is that there will always be base classes that are platform-specific, and potentially coupled with a platform-specific NativeViewHierarchyManager.

---
In reply to: [181471769](https://github.com/Microsoft/react-native-windows/pull/1772#discussion_r181471769) [](ancestors = 181471769)

Reply:
So that being said, this doesn't make me that uncomfortable, since this is tied to the "windows" base class hierarchy, which we can easily enforce that the `IReactContext` is always `ThemedReactContext`, similar to how we currently enforce that the `object`  view type is always `DependencyObject`

---
In reply to: [181481811](https://github.com/Microsoft/react-native-windows/pull/1772#discussion_r181481811) [](ancestors = 181481811,181471769)

Reply:
While I love the C# type system, this is definitely a case where generic type erasure could prove useful (so we could easily change `IViewManager` to `IViewManager<T>`

---
In reply to: [181482140](https://github.com/Microsoft/react-native-windows/pull/1772#discussion_r181482140) [](ancestors = 181482140,181481811,181471769)